stac_ipyleaflet/widgets/inspect.py

```diff
@@ -0,0 +1,129 @@
+class InspectControlWidget:
+    def template(
+        self, applied_layers, interact_widget, make_get_request, titiler_endpoint
+    ):
+        main = self
+        draw_control = DrawControl(
+            edit=False,
+            remove=False,
+            circlemarker={},
+            polygon={},
+            polyline={},
+            rectangle={},
+        )
+
+        draw_control.marker = {
+            "repeatMode": False,
+        }
+
+        interact_tab = (
+            interact_widget.children[0].children[0].children[0].children[0].children
+        )
+        point_data = interact_tab[1]
+        clear_button = interact_tab[2]
+
+        def get_visible_layers_data(coordinates) -> List[LayerData]:
+            visible_layers_data = []
+            cog_partial_request_path = (
+                f"{titiler_endpoint}/cog/point/{coordinates[0]},{coordinates[1]}?url="
+            )
+            for layer in applied_layers:
+                if "/cog" in layer.url:
+                    parsed_url = urlparse(layer.url)
+                    parsed_query = parse_qs(parsed_url.query)
+                    url = parsed_query["url"][0]
+                    data = make_get_request(f"{cog_partial_request_path}{url}")
+                    if data:
+                        visible_layers_data.append(
+                            {"layer_name": layer.name, "data": data.json()}
+                        )
+                # @TODO: Add logic to grab point data for "/mosaics" layers here
+                # @NOTE: Blocked until Impact Titiler is updated to access /mosaicsjson/point
+            return visible_layers_data
+
+        def display_data(layers_data: LayerData):
+            point_data.value = ""
+
+            def create_html_template(layer_name, coordinates, values, band_names):
+                template = f"""
+                    <p>
+                        <b>
+                            {layer_name}
+                        </b>
+                    </p>
+                    <ul>
+                        <li>
+                            Coordinates: {coordinates}
+                        </li>
+                        <li>
+                            Values: {values}
+                        </li>
+                        <li>
+                            Bands: {band_names}
+                        </li>
+                    </ul>
+                """
+                return template
+
+            for layer in layers_data:
+                point_data.value += create_html_template(
+                    layer["layer_name"],
+                    layer["data"]["coordinates"],
+                    layer["data"]["values"],
+                    layer["data"]["band_names"],
+                )
+            return
+
+        def handle_interaction(self, action, geo_json, **kwargs):
+            def handle_clear(event):
+                draw_layer = main.find_layer("draw_layer")
+                main.remove_layer(draw_layer)
```

Comment:
We should probably clear the Polygon coordinates here too - otherwise, we have ghost coordinates (values, but not indication of what they belong to because the polygon on the map has disappeared.

<img width="226" alt="Screenshot 2023-07-31 at 8 32 10 PM" src="https://github.com/MAAP-Project/stac_ipyleaflet/assets/18629881/dd2075b0-e8cd-4609-9489-1b6b278b1623">

Reply:
One additional thought is that it might be better to pull the clear button out from the tabs and handle its state, trigger it, every time there is a switch between polygon and marker.

<img width="274" alt="Screenshot 2023-07-31 at 8 32 57 PM" src="https://github.com/MAAP-Project/stac_ipyleaflet/assets/18629881/eb9403a0-029d-4bdb-bbc7-bb2f6b1f861f">

Reply:
Ghost coordinates taken care of 👍🏼 as for the additional thought, its a great one. Logic is mostly duplicated here and switching between tabs and if there is a layer with values, we should trigger on switch. I made a TODO note in the code to tackle for later as there are other TODOs in which code can be reorganized